controllers/doctors_HISP.py

```diff
@@ -1,7 +1,51 @@
-﻿from bottle import route, view, request, response
+﻿from bottle import route, view, request, response, template
 from datetime import datetime
 import sqlite3
 from permissions import permissions
+import requests, json
+
+def create_patient(url,patient):
+    # EC 112115 Expects patient dictionary and HISP address
+    # returns response to parse out id
+    query = 'base/Patient/'
+    base_url = url + query
+    # create the payload using example
+    payload = {"active": 'true',
+        "address": [{"city": "",
+        "line": [""],
+        "postalCode": "",
+        "state": "",
+        "use": ""}],
+        "birthDate": "",
+        "gender": "",
+        "id": "",
+        "name": [{"family": [""], "given": [""]}],
+        "resourceType": "",
+        "text": {"status": ""}}
+
+    payload["name"][0]["family"] = patient['lname']
+    payload["name"][0]["given"] = patient['fname']
+    payload["birthDate"] = patient['dob']
+    payload["active"] = 'true'
+    payload["resourceType"] = "Patient"
+    payload["text"] = {"status": "generated"}
+    payload["id"] = '1'
+    payload["gender"] = patient['sex']
```

Comment:
Good catch.  The FHIR server was throwing a key error when I was creating
it without initializing.  Can try to take it out bc I didn't go back to
retest it.

On Mon, Nov 23, 2015 at 4:51 PM, omtinez <notifications@github.com> wrote:

> In controllers/doctors_HISP.py
> <https://github.com/omtinez/gatech-teamfin/pull/13#discussion_r45684285>:
>
> > +        "use": ""}],
> > +        "birthDate": "",
> > +        "gender": "",
> > +        "id": "",
> > +        "name": [{"family": [""], "given": [""]}],
> > +        "resourceType": "",
> > +        "text": {"status": ""}}
> > +
> > +    payload["name"][0]["family"] = patient['lname']
> > +    payload["name"][0]["given"] = patient['fname']
> > +    payload["birthDate"] = patient['dob']
> > +    payload["active"] = 'true'
> > +    payload["resourceType"] = "Patient"
> > +    payload["text"] = {"status": "generated"}
> > +    payload["id"] = '1'
> > +    payload["gender"] = patient['sex']
>
> This seems strange to me. Why not put all the information in the payload
> variable when you create it? If it's meant as a template that can be used
> elsewhere, why not put it outside of the function?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/omtinez/gatech-teamfin/pull/13/files#r45684285>.
>